src/indicator/algorithm/calculator/envelope.js

```diff
@@ -0,0 +1,36 @@
+import d3 from 'd3';
+
+export default function() {
+
+    var channel = 0.1;
+    var midValue = function(d) {
+        return d.value;
+    };
+    var envelope = function(data) {
+        data.forEach(function(s) {
+            if (midValue(s) !== undefined) {
+                s.lowerEnvelope = midValue(s) * (1.0 - channel);
+                s.upperEnvelope = midValue(s) * (1.0 + channel);
+            }
+        });
+        return data;
+    };
+
+    envelope.channel = function(x) {
```

Comment:
is 'channel' the conventional name for this? I'd have thought that 'factor' or 'multiplier' might make more sense, although I haven't done any research on envelope indicators!